src/electrai/lightning.py

```diff
@@ -41,8 +49,16 @@ def training_step(self, batch):
 
     def validation_step(self, batch):
         x, y = batch
-        pred = self(x)
-        loss = self.loss_fn(pred, y)
+        if isinstance(x, list):
+            losses = []
+            for x_i, y_i in zip(x, y, strict=False):
+                pred = self(x_i.unsqueeze(0))
+                loss = self.loss_fn(pred, y_i.unsqueeze(0))
+                losses.append(loss)
+            loss = torch.stack(losses).mean()
+        else:
+            pred = self(x)
+            loss = self.loss_fn(pred, y)
```

Comment:
This logic is a duplicate of that in `training_step` above. It would be good to create a separate function, e.g. `_loss_calculation()` that each of these functions call. That way, next time we update this code, we won't accidentally miss an update to one and cause them to drift.